### DISCUSSION TITLE
Specifying remote package dependencies from the r-universe repo

### DISCUSSION THREAD

**Discussion by rempsyc** [r-universe-org/help#187] [Q&A] [answered]
My package [`rempsyc`](https://rempsyc.r-universe.dev/ui#package:rempsyc) has a remote GitHub dependency on the [`openxlsx2`](https://janmarvin.r-universe.dev/ui#package:openxlsx2) package.

My package GitHub actions [fail for Windows](https://github.com/rempsyc/rempsyc/runs/7622620058?check_suite_focus=true#step:5:4799) when building `openxlsx2` package from source (`Remotes: JanMarvin/openxlsx2` in `DESCRIPTION`), so the package's author thought of putting the package on the r-universe. In this way, it is already built and should be installed rather easily without leading to failed GitHub action tests.

Now I am wondering how to specify in my own package that the remote dependency is from the r-universe. I have tried the following:

```r
Suggests: 
    [...]
    remotes,
    openxlsx2,
Remotes: 
    r-universe::JanMarvin/openxlsx2
```
But it fails:
```r
 Error: Error: <callr_remote_error: Cannot install packages:
  * deps::.: Unknown remote type(s): `r-universe`>
```
Is it even possible to do what I am describing?

**Comment by jeroen** [comment]
R-universe will automatically add all external dependencies from remotes to your repository. There is nothing you need to do. I see that both `xlsx2` and `rempsyc` are available and passing in your repo: https://rempsyc.r-universe.dev/ui#builds

**Comment by rempsyc** [comment]
Oh I'm sorry, I just created my r-universe repo because I think the idea is super cool! But the issue is unrelated to my own r-universe repo (it was before I created my own r-universe repo). Rather, I am looking to get rid of the _GitHub_ actions error. If you look at my package readme on GitHub, you will see that the badge is red, indicating failing tests: https://github.com/rempsyc/rempsyc. Actually I can just paste the badge here and it works (nice, even better!):

[![R-CMD-check](https://github.com/rempsyc/rempsyc/workflows/R-CMD-check/badge.svg)](https://github.com/rempsyc/rempsyc/actions)

In order to get rid of the error, I would like GitHub actions to install the other package (`openxlsx2`) from the r-universe, and my question is essentially about how to specify that in the `DESCRIPTION` document. I have tried:
```r
Remotes: 
    r-universe::JanMarvin/openxlsx2
```
But it seems like that is not working, possibly because the `remotes` package does not recognize the r-universe as a valid repo. So, is it possible to specify the r-universe like above in the `DESCRIPTION` file for `remotes` to install it from there? Or is there any other way? Thank you.

Edit: For example, is there any way to specify repo options from _within_ the package itself so that it's picked up automatically by GitHub actions?
```
options(repos = c(
  rempsyc = 'https://rempsyc.r-universe.dev',
  CRAN = 'https://cloud.r-project.org'))
```

**Comment by rempsyc** [comment, marked as answer]
# CONFIRMED solution: 
[actions/setup-r-dependencies at v2 · r-lib/actions · GitHub](https://github.com/r-lib/actions/tree/v2/setup-r-dependencies)

> ## Dependencies in other CRAN-like repositories
>
>
>
> If your dependency is available in a CRAN-like repository (e.g. [R-universe](https://r-universe.dev/search/)), then you can use the `extra-repositories` parameter of the `setup-r` action. See more at the [`setup-r` documentation](https://github.com/r-lib/actions/tree/v2/setup-r#inputs).

So I had to add  the `extra-repositories: "https://janmarvin.r-universe.dev"` line after `with` in the `- uses: r-lib/actions/setup-r@v2` line, so it looks like this in the `R-CMD-check.yaml` workflow file:

```
  - uses: r-lib/actions/setup-r@v2
    with:
      r-version: ${{ matrix.config.r }}
      http-user-agent: ${{ matrix.config.http-user-agent }}
      use-public-rspm: true
      extra-repositories: "https://janmarvin.r-universe.dev"
```
